From ee76b399a8cb19702dde8834deb47c237d33f7bc Mon Sep 17 00:00:00 2001
From: Dave Eddy <dave@daveeddy.com>
Date: Wed, 9 May 2018 17:23:24 -0400
Subject: [PATCH] OS-6942 bhyve incapable hosts should exit early on vmadm
 create Reviewed by: Josh Wilsdon <josh@wilsdon.ca> Approved by: Josh Wilsdon
 <josh@wilsdon.ca>

---
 src/vm/node_modules/VM.js | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index d82ab6d8..416316d5 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -10036,6 +10036,35 @@ exports.create = function (payload, options, callback)
         }, function (cb) {
             var vm_type = BRAND_OPTIONS[payload.brand].features.type;
 
+            switch (vm_type) {
+            case 'BHYVE':
+                VM.getSysinfo(function gotSysinfo(err, sysinfo) {
+                    if (err) {
+                        cb(err);
+                        return;
+                    }
+
+                    if (!sysinfo.hasOwnProperty('Bhyve Capable')) {
+                        cb(new Error(
+                            'sysinfo does not have "Bhyve Capable" property'));
+                        return;
+                    }
+
+                    if (!sysinfo['Bhyve Capable']) {
+                        cb(new Error('Bhyve not supported'));
+                        return;
+                    }
+
+                    cb();
+                });
+                break;
+            default:
+                cb();
+                break;
+            }
+        }, function (cb) {
+            var vm_type = BRAND_OPTIONS[payload.brand].features.type;
+
             if (['BHYVE', 'KVM'].indexOf(vm_type) !== -1) {
                 createVM(payload, log, function (error, result) {
                     if (error) {
-- 
2.21.0

